building_sim_plugins/building_plugins_common/include/building_sim_common/crowd_simulator_common.hpp

```diff
@@ -0,0 +1,208 @@
+#ifndef BUILDING_SIM_COMMON__CROWD_SIMULATOR_COMMON_HPP
+#define BUILDING_SIM_COMMON__CROWD_SIMULATOR_COMMON_HPP
+
+#include <functional>
+#include <list>
+#include <queue>
+#include <memory>
+
+#include <MengeCore/Runtime/SimulatorDB.h>
+#include <MengeCore/Orca/ORCADBEntry.h>
+#include <MengeCore/Orca/ORCASimulator.h>
+#include <MengeCore/PluginEngine/CorePluginEngine.h>
+
+namespace crowd_simulator {
+
+using AgentPtr = std::shared_ptr<Menge::Agents::BaseAgent>;
+//===============================================================
+/*
+* class AgentPose3d
+* replace the common interface of ignition::math::Pose3d, which might apply with different version in different plugins
+*/
+class  AgentPose3d{
+
+public:
+  AgentPose3d();
+  AgentPose3d(double& x, double& y, double& z, double& pitch, double& roll, double& yaw);
+
+  ~AgentPose3d();
+
+  double X() const;
+  double Y() const;
+  double Z() const;
+  double Pitch() const;
+  double Roll() const;
+  double Yaw() const;
+
+  double& X();
+  double& Y();
+  double& Z();
+  double& Pitch();
+  double& Roll();
+  double& Yaw();
```

Comment:
I would recommend using the Setters that you have defined below for any assignment, instead of giving mutable references.